.circleci/config.yml

```diff
@@ -0,0 +1,98 @@
+    - run:
+        name: Curl
+        command: .circleci/git_no_checkin_in_last_day.sh || (cd curl && docker build -t oqs-curl . && docker run -e TEST_TIME=5 -e KEM_ALG=sikep751 -e SIG_ALG=picnicl1fs -it oqs-curl perftest.sh || echo "Test complete")
+    - run: 
+        name: Apache httpd
+        command: .circleci/git_no_checkin_in_last_day.sh || (cd httpd && docker build -t oqs-httpd-img . && docker run --detach --rm --name oqs-httpd oqs-httpd-img && sleep 2 && docker exec oqs-httpd bash -c 'echo  "GET /" | /opt/openssl/bin/openssl s_client -CAfile CA.crt -curves frodo640aes -crlf -connect localhost:4433')
+    - run: 
+        name: nginx
+        command: .circleci/git_no_checkin_in_last_day.sh || (cd nginx && docker build -t oqs-nginx-img . && docker run --detach --rm --name oqs-nginx oqs-nginx-img && sleep 2 && docker exec oqs-nginx /opt/openssl/apps/openssl s_client -CAfile CA.crt -curves kyber512 -connect localhost:4433)
+    - run:
+        name: Push images
+        command: docker tag oqs-curl $DOCKER_LOGIN/curl-generic && docker push $DOCKER_LOGIN/curl-generic && docker tag oqs-httpd-img $DOCKER_LOGIN/httpd-generic && docker push $DOCKER_LOGIN/httpd-generic && docker tag oqs-nginx-img $DOCKER_LOGIN/nginx-generic && docker push $DOCKER_LOGIN/nginx-generic
+
+
+# As of now (Mar 2020) CircleCI does not support building current OSX docker images... :-(
+#.mac_job: &macjob
+#    macos:
+#       xcode: "11.3.0"
+#    steps:
+#        - checkout # change this from "checkout" to "*localCheckout" when running CircleCI locally
+#        - run:
+#            name: Install dependencies
+#            command: |
+#              .circleci/git_no_checkin_in_last_day.sh || (
+#              brew update &&
+#              brew cask install virtualbox &&
+#              brew install docker docker-machine &&
+#              docker-machine create default &&
+#              docker-machine ls &&
+#              docker-machine start &&
+#              docker-machine env default &&
+#              eval "$(docker-machine env default)" &&
+#              docker version &&
+#              brew unlink python@2 
+#              )
+#        - run:
+#            name: docker login
+#            command: echo $DOCKER_PASSWORD | docker login --username $DOCKER_LOGIN --password-stdin
+#        - run:
+#            name: Curl
+#            command: .circleci/git_no_checkin_in_last_day.sh || (cd curl && docker build -t oqs-curl . && docker run -e TEST_TIME=5 -e KEM_ALG=sikep751 -e SIG_ALG=picnicl1fs -it oqs-curl perftest.sh)
+
+jobs:
+  debian-buster-amd64:
+    <<: *linuxjob
+    environment:
+      IMAGE: openquantumsafe/ci-debian-buster-amd64:latest
+  ubuntu-bionic-x86_64:
+    <<: *linuxjob
+    environment:
+      IMAGE: openquantumsafe/ci-ubuntu-bionic-x86_64:latest
+#  macOS:
+#    <<: *macjob
+
+workflows:
+  version: 2
+  build:
+    jobs:
+      - debian-buster-amd64:
```

Comment:
Should we make both jobs nightly ones?

Reply:
Hmm - I'd think it'd be sensible to keep one running directly at checkin to get direct feedback if all is fine. But I switched order to also validate the Ubuntu job now as we know the Debian job works.

Reply:
True. We should modify the Push images step so that we don't push new images every time a commit is pushed to a PR, perhaps using an environment variable. We can have both the jobs be nightlys that actually push images to the Docker Hub.

Reply:
Well, each job over-writes the previous' docker image if no errors occurred, so I'd argue only one should run nightly, i.e., leave as-is. Without comments to the opposite, I'll merge (say after today's call).